std/variant.d

```diff
@@ -299,7 +299,10 @@ private:
                 if (targetType != typeid(T))
                     continue;
 
-                static if (is(typeof(*cast(T*) target = *src)) ||
+                // note, variant only will ever create a new value with
+                // tryPutting (effectively), so it's OK to assume move
+                // semantics.
+                static if (is(typeof(delegate T() {return *src;})) ||
```

Comment:
Returning a value isn't a move, but a copy construction, isn't it? So you're changing the check from assignable to copy-constructible.

Reply:
I guess I don't know the correct terminology. Whatever it is where you pass an rvalue parameter to a function or return a value, that's what I was trying to say. I thought that was "move semantics".

But I can see that if `A` is not the same type as `T`, but `T` can be constructed from an `A`, it would still pass. Going to do some testing to make sure this makes sense.

Reply:
Yeah, it doesn't work if you try to get a copy-constructible type from the variant. I forgot the `T` here is some constancy flavor of `A`, not the `T` passed into `get`.

So I think it's safe to say we don't have to worry about that case.

I'll change the comment to be clearer.